gst/datarepo/gstdatareposrc.c

```diff
@@ -220,7 +185,18 @@ gst_data_repo_src_class_init (GstDataRepoSrcClass * klass)
   g_object_class_install_property (gobject_class, PROP_IS_SHUFFLE,
       g_param_spec_boolean ("is-shuffle", "Is shuffle",
           "If the value is true, samples index are shuffled",
-          DEFAULT_IS_SHUFFLE, G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+          DEFAULT_IS_SHUFFLE,
+          G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS |
+          GST_PARAM_MUTABLE_READY));
+
+  g_object_class_install_property (gobject_class, PROP_CAPS,
+      g_param_spec_boxed ("caps", "Caps",
+          "Optional property, Caps describing the format of the sample data. "
+          "Must set stop-sample-index because the total number of samples is unknown. "
```

Comment:
```Must set stop-sample-index because the total number of samples is unknown.``` for ```caps``` property doesn't appear consistent.

And, if ```stop-sample-index``` should be set, please let gstdatareposrc.c check its validity before actually starting the pipeline and write a visible and informative error message.

Reply:
okay, Thank you.